tests/js/zombie-binder-tests.js

```diff
@@ -0,0 +1,84 @@
+var path    = require("path");
+var jqUnit = require("jqUnit");
+
+fluid.registerNamespace("gpii.tests.binder");
+gpii.tests.binder.runTests = function (that) {
+    jqUnit.module("Testing binder separately from template handling...");
+    fluid.each(that.options.componentNames, function (componentName) {
+        jqUnit.asyncTest("Testing component '" + componentName + "'...", function () {
+            var browser = Browser.create();
+            browser.on("error", function (error) {
+                jqUnit.start();
+                jqUnit.fail("There should be no errors:" + error);
+            });
+            browser.visit(that.options.url, function () {
+                jqUnit.start();
+                var component = browser.window[componentName];
+                jqUnit.assertDeepEq("The initial model of component '" + componentName + "' should match the expected results.", that.options.expectedOnInit, component.model);
+
+                var formElementToUpdate = component.locate("updateFromMarkup");
+                // Zombie's `fill(selector, value)` method only works with particular types of elements, so we have to use a range of approaches for other types of form elements.
+                var tag = formElementToUpdate.prop("tagName").toLowerCase();
+                var type = formElementToUpdate.prop("type");
+                if (tag === "textarea" || (tag === "input" && type === "text")) {
+                    browser.fill(formElementToUpdate.selector, "updated using form controls");
+                }
+                // Zombie does provide to click on a radio button using only the button's value or text, but we have a number with the same value and text and want to target things more cleanly.
+                else if (tag === "input" && type === "radio") {
+                    browser.choose(formElementToUpdate.selector + "[value='updated using form controls']");
+                }
+                // Thankfully, Zombie at least allows choosing a select option matching a desired value.
+                else if (tag === "select") {
+                    browser.select(formElementToUpdate.selector, "updated using form controls");
+                }
+                jqUnit.assertEquals("A change to the form controls should result in an update to the model", "updated using form controls", component.model.updateFromMarkup);
+
+                component.applier.change("updateFromModel", "updated using applier");
+                var updatedFormElement = component.locate("updateFromModel");
+                var clientSideFluid = browser.window.fluid;
+                var clientSideValue = clientSideFluid.value(updatedFormElement);
+                jqUnit.assertEquals("A change to the model should result in an update to the form element", "updated using applier", clientSideValue);
+            });
+        });
+    });
+};
+
+var testFile = "file://" + path.resolve(__dirname, "../html/tests-binder.html");
```

Comment:
This file manipulation is not portable. These tests are failing for me on Windows with the following:

result  { module: 'Testing binder separately from template handling...',
  name: 'Testing component \'long\'...',
  result: false,
  message: 'There should be no errors:Error: Server returned status code 404 from file://e/source/gits/gpii-handlebars/tests/html/tests-binder.html',
  source: '    at pok (E:\\source\\gits\\gpii-handlebars\\node_modules\\infusion\\tests\\test-core\\jqUnit\\js\\jqUnit.js:102:15)\n    at Object.jsUnitCompat.fa
il (E:\\source\\gits\\gpii-handlebars\\node_modules\\infusion\\tests\\test-core\\jqUnit\\js\\jqUnit.js:119:13)\n    at Browser.<anonymous> (E:\\source\\gits\\gp
ii-handlebars\\tests\\js\\zombie-binder-tests.js:32:24)\n    at Browser.emit (events.js:117:20)\n    at E:\\source\\gits\\gpii-handlebars\\node_modules\\zombie\
\lib\\document.js:127:21\n    at E:\\source\\gits\\gpii-handlebars\\node_modules\\zombie\\lib\\eventloop.js:305:11\n    at Object._onImmediate (E:\\source\\gits
\\gpii-handlebars\\node_modules\\zombie\\lib\\eventloop.js:616:11)\n    at processImmediate [as _immediateCallback] (timers.js:354:15)',
  test: 'Testing component \'long\'...' }
jq:  FAIL: Module "Testing binder separately from template handling..." Test name "Testing component 'long'..." - Message: There should be no errors:Error: Serv
er returned status code 404 from file://e/source/gits/gpii-handlebars/tests/html/tests-binder.html
jq:  Source:     at pok (E:\source\gits\gpii-handlebars\node_modules\infusion\tests\test-core\jqUnit\js\jqUnit.js:102:15)
    at Object.jsUnitCompat.fail (E:\source\gits\gpii-handlebars\node_modules\infusion\tests\test-core\jqUnit\js\jqUnit.js:119:13)
    at Browser.<anonymous> (E:\source\gits\gpii-handlebars\tests\js\zombie-binder-tests.js:32:24)
    at Browser.emit (events.js:117:20)
    at E:\source\gits\gpii-handlebars\node_modules\zombie\lib\document.js:127:21
    at E:\source\gits\gpii-handlebars\node_modules\zombie\lib\eventloop.js:305:11
    at Object._onImmediate (E:\source\gits\gpii-handlebars\node_modules\zombie\lib\eventloop.js:616:11)
    at processImmediate [as _immediateCallback] (timers.js:354:15)